daemon/src/auto_rollover.rs

```diff
@@ -108,31 +108,34 @@ impl<O> Actor<O>
 where
     O: xtra::Handler<oracle::GetAnnouncement>,
 {
-    async fn handle_auto_rollover_impl(
-        &mut self,
-        ctx: &mut xtra::Context<Actor<O>>,
-    ) -> Result<(), anyhow::Error> {
+    async fn handle_auto_rollover_impl(&mut self, ctx: &mut xtra::Context<Actor<O>>) -> Result<()> {
         let mut conn = self.db.acquire().await?;
         let cfd_ids = db::load_all_cfd_ids(&mut conn).await?;
         let this = ctx
             .address()
             .expect("actor to be able to give address to itself");
         for id in cfd_ids {
-            try_continue!(async {
-                let cfd = load_cfd(id, &mut conn).await?;
-                cfd.can_auto_rollover_taker(OffsetDateTime::now_utc())?;
-
-                this.send(Rollover(id)).await??;
-
-                anyhow::Ok(())
-            }
-            .await
-            .context("Cannot roll over"));
+            load_and_trigger_rollover(id, &mut conn, &this).await?;
         }
         Ok(())
     }
 }
 
+async fn load_and_trigger_rollover<O: xtra::Handler<oracle::GetAnnouncement>>(
+    id: OrderId,
+    conn: &mut PoolConnection<Sqlite>,
+    addr: &Address<Actor<O>>,
+) -> Result<()> {
+    let cfd = load_cfd(id, conn).await?;
+    match cfd.can_auto_rollover_taker(OffsetDateTime::now_utc()) {
+        Ok(()) => {
+            addr.send(Rollover(id)).await??;
+        }
+        Err(e) => tracing::debug!("Could not auto-rollover: {:#}", e),
+    }
+    Ok(())
+}
```

Comment:
What is the purpose of extracting this function?

Matching on the result of `cfd.can_auto_rollover_taker` would also be possible within the loop above.

Also note that now, we are aborting this loop if anything fails so this is actually _less_ resilient.

Reply:
You asked to extract this block to a function in first review as far as i remember?

Reply:
I indeed did! I think that was because you just removed the `try_continue`, leaving behind an async block that was immediately awaited which looked odd.

Happy to keep the function but I think we should also keep the `try_continue`, otherwise we are not going through all CFDs if one fails.